pipeline.py

```diff
@@ -51,19 +62,28 @@ class CorpusManager:
     """
 
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = Path(path_to_raw_txt_data)
+        self._storage = {}
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        compiled_expression = re.compile(r'\d+_raw.txt')
+        for file in self.path_to_raw_txt_data.iterdir():
+            pattern = compiled_expression.match(file.name)
+            if not pattern:
+                continue
```

Comment:
there is a special pathlib method that allows you to iterate over only those files in the directory names of which satisfy a given filtering condition, so you wouldn't have to compile and match a regular expression yourself 
is there a particular reason you do not want to use it?